### PR TITLE
Generate event sub for property based tests

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description
After fixing several bugs we are finally able to have the first iteration of generating and executing property based tests with event sub processes. In this first iteration we have simple event processes with start and end event only. Furthermore event sub processes are only added on root level.

I was able to run ~2k runs without any issues.
tests

Example models which are generated:

diagram_3
diagram_6
diagram_5

111050744-bee61f80-844e-11eb-8bd8-50a733234c76
111768838-b9526480-88a8-11eb-9adc-f0c7baa2ef49
111803822-bae35300-88cf-11eb-9708-d908a69ecfc7

## Related issues

related #6195 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: engine/src/test/java/io/zeebe/engine/processing/streamprocessor/ProcessExecutionRandomizedPropertyTest.java [hunk 1]
```diff
@@ -64,6 +69,7 @@ public TestDataRecord getDataRecord() {
   @Test
   public void shouldExecuteProcessToEnd() {
     final BpmnModelInstance model = record.getBpmnModel();
+    LOG.error("Generated Model: {}", Bpmn.convertToString(model));
```
Comment (npepinpe) [inline engine/src/test/java/io/zeebe/engine/processing/streamprocessor/ProcessExecutionRandomizedPropertyTest.java:72]: Debugging left over? I guess the LOG in general as well.

File: test-util/src/main/java/io/zeebe/test/util/bpmn/random/blocks/ProcessBuilder.java [hunk 2]
```diff
@@ -56,12 +80,62 @@ public BpmnModelInstance buildProcess() {
...
+    processBuilder
+        .eventSubProcess(eventSubProcessId)
+        .startEvent("start_event_" + eventSubProcessId)
+        .interrupting(isEventSubProcessInterrupting)
+        .message(
+            b ->
+                // https://github.com/camunda-cloud/zeebe/issues/4099
+                // we might be not able to correlate a message when we have a message start event
+                // in the process
+                // as a workaround we use a constant string here
```
Comment (npepinpe) [inline test-util/src/main/java/io/zeebe/test/util/bpmn/random/blocks/ProcessBuilder.java:94]: I don't understand how the workaround works. Why is using a constant string helping? Also what if we had multiple event sub processes in the future?

**Review by npepinpe — APPROVED** [review]
:tada: 

Great stuff! No blockers, just some questions.

- Can we have a follow up issue to add event sub processes to sub processes eventually?
- If I understand correctly, we can only have one event sub process in the root? So if I want to extend it for timers, I'd need to decide on the start event, and we won't have yet multiple event sub processes?

File: test-util/src/main/java/io/zeebe/test/util/bpmn/random/blocks/ProcessBuilder.java (hunk 2 above)
Comment (Zelldon) [reply test-util/src/main/java/io/zeebe/test/util/bpmn/random/blocks/ProcessBuilder.java:94]: I think the problem is that when you have a message start event then variables are not correctly copied which will not trigger the event sub process then. If you use a static value you can trigger the event sub process.

@saig0 correct me if I'm wrong

File: engine/src/test/java/io/zeebe/engine/processing/streamprocessor/ProcessExecutionRandomizedPropertyTest.java (hunk 1 above)
Comment (Zelldon) [reply engine/src/test/java/io/zeebe/engine/processing/streamprocessor/ProcessExecutionRandomizedPropertyTest.java:72]: Ah yes sorry

File: test-util/src/main/java/io/zeebe/test/util/bpmn/random/blocks/ProcessBuilder.java [hunk 3]
```diff
@@ -88,10 +88,11 @@ private void buildEventSubProcess(
...
             b ->
-                // https://github.com/camunda-cloud/zeebe/issues/4099
-                // we might be not able to correlate a message when we have a message start event
-                // in the process
-                // as a workaround we use a constant string here
+                // When we have a message start event then variables are not correctly copied,
+                // which will not trigger the event sub process then. We use here a static value to
+                // trigger the event sub process.
+                //
+                // See https://github.com/camunda-cloud/zeebe/issues/4099
```
Comment (Zelldon) [inline test-util/src/main/java/io/zeebe/test/util/bpmn/random/blocks/ProcessBuilder.java:95]: @npepinpe does this help?

Comment (npepinpe) [reply test-util/src/main/java/io/zeebe/test/util/bpmn/random/blocks/ProcessBuilder.java:95]: Yes, thanks! :+1:

**Comment by Zelldon** [comment]
Thanks for the review @npepinpe I created some follow up issues to improve the generation further. Yes, currently we only support one event sub process on the generation.

bors r+

**Comment by npepinpe** [comment]
I tried reproducing the failing tests locally with the given seed, but they pass? Do we have some non determinism in there?

**Comment by Zelldon** [comment]
@npepinpe I think this is a race condition, since it only failed twice with the same seeds. I see in the log two terminate elements I created https://github.com/camunda-cloud/zeebe/issues/6643 I will rerun the pipeline lets see whether it fails again. If not I think we can merge to make progress here and fix it async.

**Pull request merged**